lib/services/elasticsearch.js

```diff
@@ -73,6 +73,12 @@ class ElasticSearch extends Service {
       return Bluebird.resolve(this);
     }
 
+    if (process.env.NODE_ENV === 'production' && this.config.dynamic === 'true') {
```

Comment:
I like this warning, but I'd implement this strategy the other way around. In general, I want to see my errors and the application to crash when I'm in dev mode, not production. In production my aim is stability, so I'd rather set `dynamic` to `true` when shipping and to `false` while developing.

Reply:
The application will never crash, it will only reject the request if the dynamic policy is set to `strict`.  
If the policy is `true` then the document is accepted and the mapping is updated (dangerous in production)
If the policy is `false` then the document is accepted and the mapping is not updated (best for production IMHO)
If the policy is `strict` then the document is rejected (best for development IMHO)